src/molecules/Tabs/Tabs.native.js

```diff
@@ -0,0 +1,125 @@
+import React, { useCallback, useEffect } from 'react';
+import { Dimensions } from 'react-native';
+import PropTypes from 'prop-types';
+import { TabView } from 'react-native-tab-view';
+import { getIconNames } from '../../_helpers/icon';
+import isDefined from '../../_helpers/isDefined';
+import TabBar from './TabBar';
+
+const initialLayout = { width: Dimensions.get('window').width };
+
+const Tab = () => {
+  return null;
+};
+
+Tab.propTypes = {
+  label: PropTypes.string.isRequired,
+  onPress: PropTypes.func,
+  disabled: PropTypes.bool,
+  icon: PropTypes.oneOf(getIconNames()),
+  testID: PropTypes.string,
+};
+
+Tab.defaultProps = {
+  onPress: () => {},
+  disabled: false,
+  testID: 'ds-tab-button',
+};
+
+const getTabs = ({ children }) => {
+  // Return only Tab Elements
+  return React.Children.toArray(children).filter(
+    (child) => child.type.name && child.type.name === 'Tab',
+  );
+};
+
+const getRoutes = ({ tabs }) => {
+  return tabs.map((TabComponent, index) => ({
+    index,
+    label: TabComponent.props.label,
+    key: TabComponent.props.value,
+    value: TabComponent.props.value,
+    onPress: TabComponent.props.onPress,
+    disabled: TabComponent.props.disabled,
+    icon: TabComponent.props.icon,
+    testID: TabComponent.props.testID,
+  }));
+};
+
+const getRouteIndexFromValue = ({ value, routes }) => {
+  let index = -1;
+  routes.forEach((route) => {
+    if (route.value === value) {
+      index = route.index;
+    }
+  });
+  return index;
+};
+
+const getRouteValueFromIndex = ({ index, routes }) => {
+  let value = null;
+  routes.forEach((route) => {
+    if (route.index === index) {
+      value = route.value;
+    }
+  });
+  return value;
+};
+
+const Tabs = ({ children, defaultValue, value, onChange, disableSwipe }) => {
+  const tabs = getTabs({ children });
+  const routes = getRoutes({ tabs });
+  const initialIndex = isDefined(value)
+    ? getRouteIndexFromValue({ routes, value })
+    : isDefined(defaultValue)
+    ? getRouteIndexFromValue({ routes, value: defaultValue })
+    : 0;
+  const [index, setIndex] = React.useState(initialIndex);
+
+  useEffect(() => {
+    if (isDefined(value)) setIndex(getRouteIndexFromValue({ routes, value }));
+  }, [value, routes]);
+
+  const renderScene = useCallback(
+    ({ route }) => {
+      return tabs[route.index].props.children;
+    },
+    [tabs],
+  );
+
+  const onIndexChange = useCallback(
+    (newIndex) => {
+      if (!isDefined(value)) {
```

Comment:
Do you need this check? Since you're already using `onChange` it can be part of that check itself right? i.e if(onChange) trigger the onChange else setIndex

Reply:
`setIndex` needs to happen based on whether `value` is defined even if `onChange` is defined or not. It's not if...else, it's if...if two different checks for two different purposes.

```
if (!isDefined(value)) { // if value is not defined, we can switch to a new tab but if value is defined then this is a controlled component and we should not switch to a new tab even on swipe
        setIndex(newIndex);
      }
```